prime-router/src/main/kotlin/ActionMessages.kt

```diff
@@ -140,4 +140,26 @@ class InvalidReportMessage(override val message: String) : GenericActionLogDetai
  * A [message] for invalid translation operation.
  */
 class InvalidTranslationMessage(override val message: String) :
-    GenericActionLogDetail(message, ActionLogScope.translation)
\ No newline at end of file
+    GenericActionLogDetail(message, ActionLogScope.translation)
+
+/**
+ * A [message] for entire duplicate report submission
+ */
+class DuplicateFileMessage(val payloadName: String?, val allItemsDupe: Boolean) : ActionLogDetail {
+    override val scope = ActionLogScope.report
+    override val message: String get() {
+        var msg = if (allItemsDupe) "All items in this file are duplicates." else "Duplicate file detected."
+        if (!payloadName.isNullOrEmpty()) {
+            msg += " File: $payloadName"
```

Comment:
Change the term file to maybe report or payload name (since that is the name of the URL option used).

Reply:
Updated